packages/backend/src/controllers/authentication.ts

```diff
@@ -200,6 +203,51 @@ export const googlePassportStrategy: GoogleStrategy | undefined = !(
               }
           },
       );
+
+const genericOidcHandler =
```

Comment:
Extracted the function from the okta authentication since it should work for all future OIDC authentications + fixed the first/last name logic